TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs

```diff
@@ -424,14 +424,13 @@ public static IEnumerable<CompletionItem> GetCompletionForQualifiedName(Position
                     foreach (var variable in possibleVariables)
                     {
                         var children = new List<Node>();
+                        //if it's a typed variable, we have to search for children in the type
+                        var typeChildren = GetTypeChildren(node.SymbolTable, variable);
```

Comment:
Ok with the code but the call to GetTypeChildren may be expensive (use of SymbolTable) so it would be better to check whether the call is needed or not. The special case here are the arrays which are typed and also have children (indexes).

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1605/commits/a6ceab3fa390bd3cec1b2a4f4829f41abed0104e